### PR TITLE
API Updates and NPM package system with Basic unit test cases

### PR DESCRIPTION
**Pull request by jagadeeshaby**
-  New NPM build system with gulp
-  New Mocha/Chai/Sinon bundled unit test cases with istanbul code coverage 
-  New Mute/Unmute APIs on Agent
-  Better logging 
-  Miscellaneous bug fixes

### CODE REVIEW

**Comment by flazz** [comment]
is MUTE supported in the CCP now?

**Comment by supelee** [comment]
Yes, this update will allow the agent's audio to be muted and unmuted from the front-end.

File: src/worker.js [hunk 1]
```diff
@@ -349,8 +422,8 @@
 
    ClientEngine.prototype.handleAPIRequest = function(portConduit, request) {
       var self = this;
-      var client = connect.core.getClient();
```
Comment (supelee) [inline src/worker.js]: Any reason this was changed from a clojure to a private variable?  connect.core.getClient() returns a singleton from core, so there is no overhead to doing this.

File: src/worker.js [hunk 2]
```diff
@@ -495,33 +567,62 @@
       var self = this;
       var expirationDate = new Date(self.initData.authTokenExpiration);
       var currentTimeStamp = new Date().getTime();
-      var fiveMins = 5 * 60 * 1000;
+      var thirtyMins = 30 * 60 * 1000;
```
Comment (supelee) [inline src/worker.js]: constants should be capitalized, and preferably contain the unit of the value, THIRTY_MINUTES_MS

**Comment by flazz** [comment]
@supelee I just verified the mute functionality via API, I don't see any new UX in the CCP itself, is that on its way?

File: src/worker.js (hunk 2 above)
Comment (jagadeeshaby) [reply src/worker.js]: Agreed. Its just to keep the old standards in sync  -  if its not a must have i would like to take this as part of next commit.

**Comment by extmchristensen** [comment]
@jagadeeshaby How can one test your PR while we wait for AWS to approve ? (When I try to use a modified amazon-connect-v1.2.0.js most changes are igored. My code gets loaded initially but than it switches over to a fronzen production version called lily_ccp_Prod*.gz.js at an amazon address).

**Comment by jagadeeshaby** [comment]
As discussed Internally - merging the changes to master.

**Comment by extmchristensen** [comment]
Great stuff - tiny problem is that the gulp step won't work unless you also have gulp installed globally. Better to add this section to package.json and ask users to run "npm run gulp" instead.
  **"scripts": {
    "gulp": "gulp"
  },**

**Comment by extmchristensen** [comment]
Also, there the documentation file still mentions the make file at the top under "Downloading Streams"

**Comment by extmchristensen** [comment]
Finally, the produced stream api file by gulp is misspelled and missing a verson identifier in the filename like the old file.

**Comment by jagadeeshaby** [comment]
@extmchristensen  - Thank you and Please feel free to raise an issue with the consolidated list which we will prioritize and pick it up for the implementation as soon as possible.

**Pull request merged**